src/openeo_grass_gis_driver/collection_information.py

```diff
@@ -204,47 +204,15 @@ def get(self, name):
-                    bands.append(EOBands(name="S2_3", common_name="green"))
-                    bands.append(EOBands(name="S2_4", common_name="red"))
-                    bands.append(EOBands(name="S2_5", common_name="rededge1"))
-                    bands.append(EOBands(name="S2_6", common_name="rededge2"))
-                    bands.append(EOBands(name="S2_7", common_name="rededge3"))
-                    bands.append(EOBands(name="S2_8", common_name="nir"))
-                    bands.append(EOBands(name="S2_8A", common_name="nir08"))
-                    bands.append(EOBands(name="S2_9", common_name="nir09"))
-                    bands.append(EOBands(name="S2_10", common_name="cirrus"))
-                    bands.append(EOBands(name="S2_11", common_name="swir16"))
-                    bands.append(EOBands(name="S2_12", common_name="swir22"))
-                    bands.append(EOBands(name="S2_AOT", common_name="aot"))
-                    bands.append(EOBands(name="S2_SCL", common_name="scl"))
-                    bands.append(EOBands(name="S2_TCI", common_name="tci"))
-                    bands.append(EOBands(name="S2_wvp", common_name="wvp"))
-
-                    dimensions['bands'] = {"type": "bands",
-                                           "values": ["coastal",
-                                                      "blue",
-                                                      "green",
-                                                      "red",
-                                                      "rededge1",
-                                                      "rededge2",
-                                                      "rededge3",
-                                                      "nir",
-                                                      "nir08",
-                                                      "nir09",
-                                                      "cirrus",
-                                                      "swir16",
-                                                      "swir22",
-                                                      "aot",
-                                                      "scl",
-                                                      "tci",
-                                                      "wvp"
-                                                      ]
-                                           }
+            if "band_names" in layer_data:
+                bandlist = layer_data["band_names"].split(',')
+                dimensions['bands'] = {"type": "bands",
+                                       "values": bandlist
+                                       }
+                for bandname in bandlist:
+                    # not so nice, better use different name and common_name
+                    # waiting for GRASS GIS
+                    bands.append(EOBands(name=bandname, common_name=bandname))
```

Comment:
This assumes that the band names equal always common band names, right? How many common band names are there? Could we make a list of them and if `bandname` is contained in it, we set it also for common_name and if not, it is empty or unknown?

Reply:
In this context, we only get a list of band names from `t.info`. In TGIS, there is only a single field `band_reference`, thus we get only one band name per raster and use it as both name and common_name.
A list of common band names in the EO STAC extension is here: https://github.com/stac-extensions/eo/#common-band-names

Reply:
Yes - I meant that it can be the case that the user specifies a `band_reference` which is not a common band name but nevertheless it would appear in this field then. To prevent this, a check could be implemented to compare the field with all common band names and only set it then, otherwise leave it empty.